FishBun/src/main/java/com/sangcomz/fishbun/adapter/view/PickerGridAdapter.java

```diff
@@ -70,8 +70,12 @@ public void onBindViewHolder(final RecyclerView.ViewHolder holder, final int pos
             vh.header.setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    if (pickerController.checkCameraPermission()) {
-                        pickerController.takePicture((Activity) vh.header.getContext(), saveDir);
+                    if (pickerController.isCameraPermissionDeclared()) {
+                        if (pickerController.checkCameraPermission()) {
+                            pickerController.takePicture((Activity) vh.header.getContext(), saveDir);
+                        }
+                    } else {
+                        pickerController.showDeclarePermissionDialog();
```

Comment:
I think it's better to call it only once when calling `CheckCameraPermission` from `PermissionCheck` without calling isCameraPermissionDeclared here and albumActivity.

Reply:
If i return it yes it will crash bcoz you are opening the camera without permission so it will throw some securityexception or denial exception.

Also if a programmer doesn't want camera permission in his app then he will setCamera(false).

I think no one will use setCamera(true) and then forget to add camera permission

Reply:
> I think it's better to call it only once when calling `CheckCameraPermission` from `PermissionCheck` without calling isCameraPermissionDeclared here and albumActivity.

Ok i can merge the methods this is doable

Reply:
> > I think it's better to call it only once when calling `CheckCameraPermission` from `PermissionCheck` without calling isCameraPermissionDeclared here and albumActivity.
> 
> Ok i can merge the methods this is doable

Sorry its not doable bcoz we need to show different message in case of permission not declared in project. Otherwise it will say permission denied like before

Reply:
> If i return it yes it will crash bcoz you are opening the camera without permission so it will throw some securityexception or denial exception.
> 
> Also if a programmer doesn't want camera permission in his app then he will setCamera(false).
> 
> I think no one will use setCamera(true) and then forget to add camera permission

FishBun doesn't require camera permissions by default. This is because the camera is used with `ACTION_IMAGE_CAPTURE`. 
The situation where FishBun needs to request camera permissions is when an app using FishBun declares a camera permission in the manifest.
So if your app declares camera permissions in the manifest, then FishBun will check the camera permissions before using the camera. If not, FishBun will request the intent and use the camera.

Reply:
Ok understood i will make the necessary changes and test the flow once

Reply:
I have done all the required changes and the flow is working fine for the below cases :

**1. If camera permission exists in manifest**
It checks for permission and then show the runtime permission dialog and the process continues as normal behavior.

**2. If camera permission doesn't exists in manifest**

It just show the camera intent and it works flawlessly